pandas/io/pytables.py

```diff
@@ -2712,10 +2712,11 @@ def write_array(self, key, value, items=None):
             raise NotImplementedError('Cannot store a category dtype in '
                                       'a HDF5 dataset that uses format='
                                       '"fixed". Use format="table".')
-
         if not empty_array:
-            value = value.T
-            transposed = True
+            if hasattr(value, 'T'):
+                # ExtensionArrays (1d) may not have transpose.
```

Comment:
ndarray[ndim=1], Series, and Index have `transpose` that returns a view on `self`.  Any reason not to do the same for EA?  (I tentatively think adding this behavior could reduce a bunch of boilerplate i tests.arithmetic)

Reply:
For a 1-D ndarray, `.transpose` makes sense because it's a special case of the n-dimensional concept.

I don't think it's necessary for ExtensionArrays, which are 1-dimensional by definition.

Does a similar `getattr(obj, 'T', obj)` not work to deduplication boilerplate?

Reply:
> Does a similar getattr(obj, 'T', obj) not work to deduplication boilerplate?

 It would work, just slightly less pretty.

> I don't think it's necessary for ExtensionArrays, which are 1-dimensional by definition.

ATM ndarray/Index/Series behave one way and EA behaves another.  This isn't all that big a deal, but ironing out special cases so we (and users) don't have to check which type of object they have is convenient.

Reply:
I'd like to balance that against not overloading the EA API with unnecessary methods.

I wouldn't consider this change in `pytables.py` a long-lasting one. Once we add proper support for serializing ExtensionArrays, we'll dispatch to a method on the EA that will be responsible for returning the right dtype / shaped thing.

Reply:
let's make an issue about the serializing, need to add some protocol to EA base I think.